### PR TITLE
Temporary disable GVNIC since DLVM images do not support GVNIC.

### PR DESCRIPTION
**Pull request by soumyapani**

### CODE REVIEW

File: aiinfra-cluster/modules/vm-instance-group/main.tf [hunk 1]
```diff
@@ -25,6 +25,12 @@ locals {
   enable_gvnic  = var.bandwidth_tier != "not_enabled"
   enable_tier_1 = var.bandwidth_tier == "tier_1_enabled"
 
+  is_debian_11_image = (var.instance_image.name != null && length(regexall("debian-11", var.instance_image.name)) > 0) || (var.instance_image.family != null && length(regexall("debian-11", var.instance_image.family)) > 0)
```
Comment (abdallag) [inline aiinfra-cluster/modules/vm-instance-group/main.tf:28]: Should we support non debian image families (e.g. centos)

**Review by DmitryKakurin — APPROVED** [review]

**Review by abdallag — APPROVED** [review]

File: aiinfra-cluster/modules/vm-instance-group/main.tf [hunk 2]
```diff
@@ -25,6 +25,12 @@ locals {
...
+  
+  // Terraform does not provide a way to validate multiple variables in variable validation block.
+  // Using this type of validation as per https://github.com/hashicorp/terraform/issues/25609#issuecomment-1057614400
+  validate_basic_node_pool = (!local.is_debian_11_image && (local.enable_gvnic || local.enable_tier_1)) ? tobool("GVNIC is only supported in debian 11 image.") : true
```
Comment (stevenBorisko) [inline aiinfra-cluster/modules/vm-instance-group/main.tf:32]: this is clever!

should the error message also include something about `enable_tier_1` or is that implied in `GVNIC`?

**Review by stevenBorisko — APPROVED** [review]

File: aiinfra-cluster/modules/vm-instance-group/main.tf (hunk 2 above)
Comment (soumyapani) [reply aiinfra-cluster/modules/vm-instance-group/main.tf:32]: enable_tier_1 is GVNIC + tier_1. So GVNIC is implied with that.

File: aiinfra-cluster/modules/vm-instance-group/main.tf (hunk 1 above)
Comment (soumyapani) [reply aiinfra-cluster/modules/vm-instance-group/main.tf:28]: Currently we are limiting to dlvm and I didn't see any dlvm centos image. As we expand out of DLVM, we can start supporting centos, but that is a separate project by itself.

```
centos-7-v20230306                                      centos-cloud         centos-7                                       READY
centos-stream-8-v20230306                               centos-cloud         centos-stream-8                                READY
centos-stream-9-v20230306                               centos-cloud         centos-stream-9                                READY
```

**Pull request merged**